tracing-subscriber/src/registry/stack.rs

```diff
@@ -0,0 +1,60 @@
+use std::cell::RefCell;
+use std::collections::HashSet;
+
+pub(crate) use tracing_core::span::Id;
+
+struct ContextId {
+    id: Id,
+    duplicate: bool,
+}
+
+/// `SpanStack` tracks what spans are currently executing on a thread-local basis.
+///
+/// A "separate current span" for each thread is a semantic choice, as each span
+/// can be executing in a different thread.
```

Comment:
i'm not sure if i would say that this is a semantic choice we're making here in particular as much as it's the generally expected semantics for spans...not going to block on a comment on an internal api, though.